src/AutoRest.CSharp/Common/Output/Builders/SerializationMethodsBuilder.cs

```diff
@@ -45,7 +45,9 @@ public static IEnumerable<Method> BuildSerializationMethods(SerializableObjectTy
             {
                 if (model.IncludeSerializer)
                 {
-                    foreach (var method in JsonSerializationMethodsBuilder.BuildJsonSerializationMethods(json, serialization.Interfaces))
+                    bool hasInherits = model.Inherits != null && model.Inherits is { IsFrameworkType: false };
```

Comment:
`model.Inherits is {IsFrameworkType: false}` implies `model.Inherits is not null`
therefore I think this could be simplified.